src/components/questionnaire.js

```diff
@@ -0,0 +1,224 @@
+        this.startQuestionnaire();
+    };
+
+	getInitialState = () => {
+        return {
+            activeMessage: {},
+			answers: {},
+			validationErrors: [],
+            log: [],
+			isLoading: false,
+			messages: this.props.messages
+        };
+	};
+	
+	addToLog = (item) => {
+		let log = this.state.log;
+		log.push(item);
+		this.setState({
+			log: log
+		});
+	};
+
+	getLogById = (id) => {
+		return this.state.log.filter(item => Object.hasOwnProperty.call(item, 'id') ? item.id === id : false)[0];
+	};
+
+	getMessageById = (id) => {
+		if (!id) {
+			return this.state.messages[0];
+		}
+		return this.state.messages.filter(message => message.id === id)[0];
+	};
+
+	getNextMessage = (nextStep, callback) => {
+		// Get the next message
+		const nextMessage = this.getMessageById(nextStep);
+
+		// Push the question to the log
+		this.addToLog(nextMessage);
+	};
+
+	getValidationErrors = (validationTypes, input, key) => {
+        let validationErrors = [];
+        const validator = new Validator();
+
+        for (let validationType of validationTypes) {
+            var validationObj = validator.Rules[validationType];
+
+            if (!validationObj.test(input)) {
+                validationErrors.push({
+                    type: validationType,
+                    input: input,
+                    message: validationObj.message
+                })
+            }
+		}
+
+		var newErrors = this.state.validationErrors;
+		newErrors = newErrors.concat(validationErrors);
+
+		if (validationErrors.length) {
+			this.setState({
+				validationErrors: newErrors
+			})
+		}
+		else {
+			this.setState({
+				validationErrors: []
+			});
+		}	
+		return validationErrors;
+	};
+
+	handleButtonSelect = (questionId, answerValue, answerInfo) => {
+		let activeQuestion = this.getMessageById(questionId);
+
+		if (!answerInfo) {
+			answerInfo = {};
+
+			// The answer must already be in the state
+			const answerObjExists = Object.hasOwnProperty.call(this.state.answers, answerValue);
+			const answer = answerObjExists && !!this.state.answers[answerValue] 
+				? this.state.answers[answerValue] : null;
+
+			if (!answerObjExists || !answer) {
+				return;
+			}
+
+			// Set the step = to the value in the state
+			answerInfo.text = answer;
+
+			//What is the next Step
+			// A custom method or the options should figure out what this is
+			var isMethod = typeof activeQuestion.options === 'function';
+
+			if (isMethod) {
+				answerInfo.nextStep = activeQuestion.options(this.state.answers);
+			}
+		}
+
+		// Set the active question to answered
+		this.setLogQuestionToAnswered(questionId);
+
+		// Push the answer to the question log
+		this.addToLog({
+			text: answerInfo.text,
+			type: 'answer'
+		});
+
+		// Add the answer to the answer list
+		this.setAnswer(activeQuestion.key, answerValue);
+
+		// Go to the next question, if it exists
+		this.getNextMessage(answerInfo.nextStep);
+	};
+
+	handleFormSubmit = (e) => {
+		console.log(e);
```

Comment:
I'm just gonna go out on a limb here and suggest you do something more on submit :)